app/controllers/reservations_controller.rb

```diff
@@ -295,10 +304,9 @@ def ability_resource
 
   # TODO you shouldn't be able to edit reservations that have passed or are outside of the cancellation period (check to make sure order has been placed)
   def invalid_for_update?
-    params[:id].to_i != @reservation.id ||
-    @reservation.actual_start_at ||
-    @reservation.actual_end_at ||
-    !(@reservation.can_edit? || @reservation.can_customer_edit?)
+    different_id = params[:id].to_i != @reservation.id
+    cannot_edit = current_user.administrator? ? !@reservation.can_edit? : !@reservation.can_customer_edit?
```

Comment:
I removed  the check for `@reservation.actual_start_at` because if a reservation has been started, then it will have `@reservation.actual_start_at` which would make `invalid_for_update?` return true.

I can probably re-add the check for `@reservation.actual_end_at` because we want to keep that.

For the abilities check, maybe we can work through this together because I want to make sure I do it right.